components/webapp-mgt/org.wso2.carbon.webapp.mgt/src/main/java/org/wso2/carbon/webapp/mgt/loader/CarbonWebappClassLoader.java

```diff
@@ -46,26 +46,18 @@
     private static final Log log = LogFactory.getLog(CarbonWebappClassLoader.class);
 
     private WebappClassloadingContext webappCC;
-
     private static List<String> systemPackages;
     private static final String CLASS_FILE_SUFFIX = ".class";
 
     public CarbonWebappClassLoader(ClassLoader parent) {
-        super(parent);
+        super(new SharedURLClassLoader(new URL[0], parent));
```

Comment:
Should not we use the Shared Class loader factory here. Cration a new sharecURL classloader seems counterintuitive